### PR TITLE
Mercury tests fixes

### PR DESCRIPTION
**Pull request by nikodamn**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/auth/ForcedLoginTests.java [hunk 1]
```diff
@@ -46,11 +46,18 @@ public void anonCanLogInViaUserLoginPage() {
     assertTrue(base.isStringInURL(URLsContent.SPECIAL_UPLOAD));
   }
 
+  @Test(invocationCount = 5)
   public void anonCanLogInOnSpecialWatchListPage() {
     WikiBasePageObject base = new WikiBasePageObject();
     base.openSpecialWatchListPage(wikiURL);
     base.clickLoginOnSpecialPage();
     new AttachedSignInPage().login(user);
+    //TODO: To delete
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/auth/ForcedLoginTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)

File: src/test/java/com/wikia/webdriver/testcases/auth/ForcedLoginTests.java [hunk 2]
```diff
@@ -46,11 +46,18 @@ public void anonCanLogInViaUserLoginPage() {
...
+    try {
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/auth/ForcedLoginTests.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java [hunk 3]
```diff
@@ -229,12 +230,17 @@ private RegisterPage navigateToSignUpOnMobile(TopBar topBar) {
       .navigateToSignUp();
   }
 
-  private RegisterPage openSignUpModalOnDesktop() {
-    return new DetachedRegisterPage(new NavigationBar().clickOnRegister());
+//  private RegisterPage openSignUpModalFromGlobalavOnDesktop() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/auth/register/AttachedRegisterPage.java [hunk 4]
```diff
@@ -42,7 +42,12 @@ public AttachedRegisterPage() {
 
   @Override
   public AttachedRegisterPage open() {
-    driver.get(urlBuilder.getUrl() + URLsContent.USER_SIGNUP);
+    try {
+      driver.get(urlBuilder.getWikiGlobalURL() + URLsContent.USER_SIGNUP + "?redirect=" + URLEncoder
+          .encode(urlBuilder.getUrl(), "UTF-8"));
+    } catch (UnsupportedEncodingException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/auth/register/AttachedRegisterPage.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/auth/register/AttachedRegisterPage.java [hunk 5]
```diff
@@ -42,7 +42,12 @@ public AttachedRegisterPage() {
...
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/auth/register/AttachedRegisterPage.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/testcases/auth/LoginTests.java [hunk 6]
```diff
@@ -230,12 +231,17 @@ private SignInPage navigateToSignInOnMobile(TopBar topBar) {
       .navigateToSignIn();
   }
 
-  private SignInPage openLoginModalOnDesktop() {
-    return new DetachedSignInPage(new NavigationBar().clickOnSignIn());
+//  private SignInPage openLoginModalOnDesktop() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/auth/LoginTests.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/testcases/auth/ForcedLoginTests.java [hunk 7]
```diff
@@ -5,6 +5,7 @@
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.core.api.ArticleContent;
 import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.logging.Log;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/auth/ForcedLoginTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.logging.Log'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by nikodamn** [comment]
Sonar build passed on #8032. Not sure why it's not visible here.

**Pull request merged**